apikit-tools/apikit-maven-plugin/src/main/java/org/mule/tools/apikit/CreateMojo.java

```diff
@@ -97,11 +103,19 @@ public void execute()
 
         List<String> specFiles = getIncludedFiles(specDirectory, specIncludes, specExcludes);
         List<String> muleXmlFiles = getIncludedFiles(muleXmlDirectory, muleXmlIncludes, muleXmlExcludes);
-
+        String domainFile = null;
+        if (domainDirectory != null)
+        {
+            List<String> domainFiles = getIncludedFiles(domainDirectory, muleXmlIncludes, muleXmlExcludes);
+            if (domainFiles.size() > 0)
+            {
+                domainFile = domainFiles.get(0);
```

Comment:
You could add an info log saying that there was more than one domain file and that the scaffolder is going to use the first one, and log which one that is.